corehq/apps/locations/forms.py

```diff
@@ -333,38 +337,60 @@ def save(self, instance=None, commit=True):
         return location
 
 
-class UsersAtLocationForm(MultipleSelectionForm):
+class UsersAtLocationForm(forms.Form):
+    selected_ids = forms.Field(
+        label=ugettext_lazy("Group Membership"),
+        required=False,
+        widget=Select2Ajax(multiple=True),
+    )
 
     def __init__(self, domain_object, location, *args, **kwargs):
         self.domain_object = domain_object
         self.location = location
+        fieldset_title = kwargs.pop('fieldset_title',
+                                    ugettext_lazy("Edit Group Membership"))
+        submit_label = kwargs.pop('submit_label',
+                                  ugettext_lazy("Update Membership"))
+
         super(UsersAtLocationForm, self).__init__(
             initial={'selected_ids': self.users_at_location},
             *args, **kwargs
         )
-        self.fields['selected_ids'].choices = self.get_all_users()
-        self.fields['selected_ids'].label = ugettext_lazy("Workers at Location")
-
-    def get_all_users(self):
-        user_query = (UserES()
-                      .domain(self.domain_object.name)
-                      .mobile_users()
-                      .fields(['_id', 'username', 'first_name', 'last_name']))
-        return [
-            (u['_id'], user_display_string(u['username'],
-                                           u.get('first_name', ''),
-                                           u.get('last_name', '')))
-            for u in user_query.run().hits
-        ]
+
+        from corehq.apps.reports.filters.api import MobileWorkersOptionsView
+        self.fields['selected_ids'].widget.set_url(
+            reverse(MobileWorkersOptionsView.urlname, args=(self.domain_object.name,))
+        )
+        self.fields['selected_ids'].widget.set_initial(self.users_at_location)
+        self.helper = FormHelper()
+        self.helper.label_class = 'col-sm-3 col-md-2'
+        self.helper.field_class = 'col-sm-9 col-md-8 col-lg-6'
+        self.helper.form_tag = False
+
+        self.helper.layout = crispy.Layout(
+            crispy.Fieldset(
+                fieldset_title,
+                crispy.Field('selected_ids'),
+            ),
+            hqcrispy.FormActions(
+                crispy.ButtonHolder(
+                    Submit('submit', submit_label)
+                )
+            )
+        )
 
     @property
     @memoized
     def users_at_location(self):
-        return (UserES()
-                .domain(self.domain_object.name)
-                .mobile_users()
-                .location(self.location.location_id)
-                .get_ids())
+        user_query = UserES().domain(
```

Comment:
Formatting looks a little weird here. the convention is to use something similar to what it was before

```
(UserES()
 .domain(domain).
 .filter(...))
```